test/SPIRV/opundef.ll

```diff
@@ -0,0 +1,37 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t2.bc
+; RUN: llvm-dis < %t2.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-LLVM: entry:
+; CHECK-LLVM: %agg1 = insertvalue %{{[0-9]*}} undef
+; CHECK-LLVM: ret void
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir64-unknown-unknown"
+
+%0 = type { i32, float }
+
+; Function Attrs: nounwind
+define spir_func void @foo() #0 {
+entry:
+  %agg1 = insertvalue %0 undef, i32 1, 0

```

Comment:
can you add a check that undef instruction is inside a SPIR-V function? I have some doubt about that. As I remember, the translator will generate undef as constant outside of any functions. If that's the case, we may need a hand-crafted SPIR-V module. Also, better to have two functions each containing an undef instruction of the same type.